File: src/main/java/com/github/xumeng/java/problems/easy/problem0345/Solution0345.java
package com.github.xumeng.java.problems.easy.problem0345;

public class Solution0345 {

    public String reverseVowels(String s) {
        char[] characters = s.toCharArray();
        char temp;
        for (int i = 0, j = characters.length - 1; i < j; ) {
            if (isNotVowel(characters[i])) {
                i++;
                continue;
            }
            if (isNotVowel(characters[j])) {
                j--;
                continue;
            }
            temp = characters[i];
            characters[i] = characters[j];
            characters[j] = temp;
            i++;
            j--;
        }
        StringBuilder builder = new StringBuilder();
        for (char character : characters) {
            builder.append(character);
        }
        return builder.toString();
    }

    private boolean isNotVowel(char character) {
        return switch (character) {
            case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' -> false;
            default -> true;
        };
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0345/Solution0345.kt
package com.github.xumeng.kotlin.problems.easy.problem0345

class Solution0345 {

    fun reverseVowels(s: String): String {
        val characters = s.toCharArray()
        var temp: Char
        var i = 0
        var j = characters.size - 1
        while (i < j) {
            if (isNotVowel(characters[i])) {
                i++
                continue
            }
            if (isNotVowel(characters[j])) {
                j--
                continue
            }
            temp = characters[i]
            characters[i] = characters[j]
            characters[j] = temp
            i++
            j--
        }
        return String(characters)
    }

    private fun isNotVowel(character: Char): Boolean {
        return when (character) {
            'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' -> false
            else -> true
        }
    }

}